src/withPseudoState.js

```diff
@@ -37,26 +37,37 @@ const shadowHosts = new Set()
 addons.getChannel().on(STORY_CHANGED, () => shadowHosts.clear())
 
 // Global decorator that rewrites stylesheets and applies classnames to render pseudo styles
-export const withPseudoState = (StoryFn) => {
-  const parameter = useParameter("pseudo")
+export const withPseudoState = (StoryFn, { viewMode, parameters, id }) => {
+  const isDocs = viewMode === 'docs';
+  const { pseudo: parameter } = parameters;
   const [{ pseudo: globals }, updateGlobals] = useGlobals()
 
-  // Sync parameter to globals, used by the toolbar
+  // Sync parameter to globals, used by the toolbar (only in canvas as this
+  // doesn't make sense for docs because many stories are displayed at once)
   useEffect(() => {
-    if (parameter !== globals) updateGlobals({ pseudo: parameter })
-  }, [parameter])
+    if (parameter !== globals && !isDocs) updateGlobals({ pseudo: parameter })
+  }, [parameter, isDocs])
 
   // Convert selected states to classnames and apply them to the story root element.
   // Then update each shadow host to redetermine its own pseudo classnames.
   useEffect(() => {
-    applyClasses(
-      document.getElementById("root"),
-      Object.entries(globals || {})
-        .filter(([_, value]) => value)
-        .map(([key]) => `pseudo-${PSEUDO_STATES[key]}`)
-    )
-    shadowHosts.forEach(updateShadowHost)
-  }, [globals])
+    const apply = element => {
+      applyClasses(
+        element,
+        Object.entries(parameter || {})
+          .filter(([_, value]) => value)
+          .map(([key]) => `pseudo-${PSEUDO_STATES[key]}`)
+      )
+      shadowHosts.forEach(updateShadowHost)
+    }
+
+    if (isDocs) {
+      // Wait for the docs page to render so we can select the story element 😑
+      setTimeout(() => apply(document.getElementById(`story--${id}`)), 0);
```

Comment:
This is an unreliable technique. I'm unfamiliar with SB internals at all, so don't know whether there's a lifecycle event/hook/API to capture the exact moment in time when it happens. If there isn't one, I'd use `IntersectionObserver` for this purpose.

Reply:
I think this is fine for now, but good to keep in mind.